src/jsonrpc.rs

```diff
@@ -0,0 +1,62 @@
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+#[derive(Debug, PartialEq, Clone, Hash, Eq, Deserialize, Serialize, PartialOrd, Ord)]
+#[serde(deny_unknown_fields)]
+#[serde(untagged)]
+pub enum Id {
+    /// Null
+    Null,
+    /// Numeric id
+    Number(u64),
+    /// String id
+    Str(String),
+}
+
+/// JSON-RPC request object as defined in the [spec](https://www.jsonrpc.org/specification#request-object).
```

Comment:
I don't think we need to do this now, but it may be worth considering if we can leverage an existing jsonrpc crate, such as https://docs.rs/jsonrpc.

Reply:
Yeah, I looked at jsonrpsee, but I didn't want to drag in another dependency right now. Might be worth it long term, though.